x-pack/plugins/threat_intelligence/public/modules/query_bar/components/filter_in/filter_in.tsx

```diff
@@ -0,0 +1,94 @@
+ * 2.0.
+ */
+
+import React, { useCallback, VFC } from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiButtonEmpty, EuiButtonIcon, EuiContextMenuItem } from '@elastic/eui';
+import { Filter } from '@kbn/es-query';
+import { useIndicatorsFiltersContext } from '../../../indicators/hooks/use_indicators_filters_context';
+import { getIndicatorFieldAndValue } from '../../../indicators/lib/field_value';
+import { FilterIn as FilterInConst, updateFiltersArray } from '../../lib/filter';
+import { EMPTY_VALUE } from '../../../../../common/constants';
+import { Indicator } from '../../../../../common/types/indicator';
+import { useStyles } from './styles';
+
+export const ICON_TEST_ID = 'tiFilterInIcon';
+export const COMPONENT_TEST_ID = 'tiFilterInComponent';
+
+const ICON_TYPE = 'plusInCircle';
+const ICON_TITLE = i18n.translate('xpack.threatIntelligence.queryBar.filterInButton', {
+  defaultMessage: 'Filter In',
+});
+
+export interface FilterInProps {
+  /**
+   * Value used to filter in/out in the KQL bar. Used in combination with field if is type of {@link Indicator}.
+   */
+  data: Indicator | string;
+  /**
+   * Value used to filter in /out in the KQL bar.
+   */
+  field: string;
+  /**
+   * Display component for when the FilterIn component is used within a DataGrid
+   */
+  Component?: typeof EuiButtonEmpty | typeof EuiButtonIcon;
+  /**
+   * For when FilterIn component is used within a context menu
+   */
+  contextMenu?: boolean;
+}
+
+/**
+ * Retrieves the indicator's field and value, then creates a new {@link Filter} and adds it to the {@link FilterManager}.
+ */
+export const FilterIn: VFC<FilterInProps> = ({ data, field, Component, contextMenu }) => {
+  const styles = useStyles();
+
+  const { filterManager } = useIndicatorsFiltersContext();
+
+  const { key, value } =
+    typeof data === 'string' ? { key: field, value: data } : getIndicatorFieldAndValue(data, field);
+
+  const filterIn = useCallback((): void => {
+    const existingFilters = filterManager.getFilters();
+    const newFilters: Filter[] = updateFiltersArray(existingFilters, key, value, FilterInConst);
+    filterManager.setFilters(newFilters);
+  }, [filterManager, key, value]);
+
+  if (!value || value === EMPTY_VALUE || !key) {
+    return <></>;
+  }
+
+  if (Component) {
+    return (
+      <div data-test-subj={COMPONENT_TEST_ID} css={styles.button}>
+        <Component aria-label={ICON_TITLE} iconType={ICON_TYPE} onClick={filterIn} />
+      </div>
+    );
+  }
+
+  if (contextMenu) {
+    return (
+      <EuiContextMenuItem key="addToTimeline" icon="plusInCircle" size="s" onClick={filterIn}>
+        Filter In
+      </EuiContextMenuItem>
+    );
+  }
+
+  return (
+    <EuiButtonIcon
+      data-test-subj={ICON_TEST_ID}
+      aria-label={ICON_TITLE}
+      iconType={ICON_TYPE}
+      iconSize="s"
+      size="xs"
+      color="primary"
+      onClick={filterIn}
+    />
+  );
+};
```

Comment:
I'm not sure this approach is the best to be honest... I'm only adding one more `if` condition to what was there before, but for some reason I'm not a fan of it...

Reply:
I don't see the problem with the `if` itself. My concern is that probably the component shouldn't know about the context it is used. I mean we have one component with two states "just icon" and "icon with text". It doesn't really matter if the latter is used in the context menu, it can be as well present anywhere on the page. It just has text. So i'd find a suitable name for the prop, eg. 'withActionText' or smth

Reply:
I like that idea, the only _problem_ I see with this `withActionText` option is that for a use within a `EuiContextMenu` I need to use a `EuiContextMenuItem` component.
The use of a `withActionText` boolean could work but it wouldn't be obvious what the component does with this name. I'll keep thinking about it...

Reply:
I like this, I can even use the type for the EuiGrid instead of checking against the existence of `Component`. Awesome thank you!
=